hnn_core/network.py

```diff
@@ -872,6 +874,11 @@ def _attach_drive(self, name, drive, weights_ampa, weights_nmda, location,
             raise ValueError('Allowed drive target cell types are: ',
                              f'{self.cell_types.keys()}')
 
+        # enforce the same order as in self.cell_types - necessary for
+        # consistent source gid assignment
+        target_populations = [cell_type for cell_type in self.cell_types.keys()
+                              if cell_type in target_populations]
```

Comment:
This was the real issue + solution: the connections from drive source gids to target cells weren't being instantiated in the same order every time. The convention I adopted is to always use the order specified in `Network.cell_types`.

Reply:
That is such a good find